stacker/blueprints/info.py

```diff
@@ -0,0 +1,244 @@
+from .variables.types import (
+    CFNType,
+    TroposphereType,
+)
+from ..util import load_object_from_string
+
+
+def format_var_type(var_type):
+    """Format a given var type to be displayed by the info command.
+
+    Args:
+        var_type (type): type of variable
+
+    Returns:
+        str: formatted display variable type
+
+    """
+    is_list = False
+    if isinstance(var_type, TroposphereType):
+        if var_type.many:
+            is_list = True
+
+        var_type = var_type.type
+
+    if isinstance(var_type, (list, tuple)) and len(var_type):
+        is_list = True
+        var_type = var_type[0]
+
+    if isinstance(var_type, CFNType):
+        parts = ["{}::{}".format("CFN", var_type.parameter_type)]
+    else:
+        parts = [var_type.__name__]
+        module = getattr(var_type, "__module__", None)
+        if module == "__builtin__":
+            module = None
+
+        if module:
+            parts.insert(0, module)
+
+    formatted_type = ".".join(parts)
+    return "[{}]".format(formatted_type) if is_list else formatted_type
+
+
+def get_troposphere_type(var_type):
+    """Return the wrapped type from a TroposphereType.
+
+    Args:
+        var_type (:class:`stacker.blueprints.variables.types.TroposphereType`):
+            wrapped type
+
+    Returns:
+        type: wrapped inner troposphere type
+
+    """
+    troposphere_type = var_type.type
+    if isinstance(var_type.type, list):
+        troposphere_type = var_type.type[0]
+    return troposphere_type
+
+
+def explain_variable_field(variable, field_path):
+    """Format information for a field of a given variable.
+
+    Args:
+        variable (dict): blueprint variable
+        field_path (list): path to the field within the variable
+
+    Returns:
+        str: formatted string describing the field for the given variable.
+
+    """
+    var_type = variable["type"]
+    # TODO does this handle cases where we don't have a troposphere type here?
+    troposphere_type = get_troposphere_type(var_type)
+
+    while field_path:
+        prop_name = field_path.pop(0)
+        prop_type, required = troposphere_type.props[prop_name]
+        if isinstance(prop_type, list) and len(prop_type) == 1:
+            prop_type = prop_type[0]
+
+        if hasattr(prop_type, "props"):
+            troposphere_type = prop_type
+
+    fields = ""
+    description = [format_var_type(prop_type)]
+    if not hasattr(prop_type, "props"):
```

Comment:
This is all getting a little complicated - maybe it'd be good to break it up into smaller functions, or possibly just throw some comments in here?